rocAL/rocAL_pybind/amd/rocal/readers.py

```diff
@@ -22,158 +22,184 @@
 from amd.rocal.pipeline import Pipeline
 import amd.rocal.types as types
 
-def coco(*inputs, file_root, annotations_file='', bytes_per_sample_hint=0, dump_meta_files=False,
-         dump_meta_files_path='', file_list='', initial_fill=1024,  lazy_init=False, ltrb=False, masks=False,
-         meta_files_path='', num_shards=1, pad_last_batch=False, prefetch_queue_depth=1, preserve=False,
-         random_shuffle=False, ratio=False, read_ahead=False, save_img_ids=False, seed=-1, shard_id=0,
-         shuffle_after_epoch=False, size_threshold=0.1, skip_cached_images=False, skip_empty=False,
-         stick_to_shard=False, tensor_init_bytes=1048576, is_box_encoder=False, is_box_iou_matcher=False):
 
+def coco(annotations_file='', ltrb=True, masks=False, ratio=False,
+         avoid_class_remapping=False, pixelwise_masks=False, is_box_encoder=False, is_box_iou_matcher=False):
     Pipeline._current_pipeline._reader = "COCOReader"
-    #Output
+    # Output
     labels = []
     bboxes = []
-    kwargs_pybind = {"source_path": annotations_file, "is_output":True, "is_box_encoder":is_box_encoder, "is_box_iou_matcher":is_box_iou_matcher }
-    meta_data = b.cocoReader(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+    kwargs_pybind = {
+        "source_path": annotations_file,
+        "is_output": True,
+        "mask": masks,
+        "ltrb": ltrb,
+        "is_box_encoder": is_box_encoder}
+    meta_data = b.cocoReader(Pipeline._current_pipeline._handle, *(kwargs_pybind.values()))
     return (meta_data, labels, bboxes)
 
-def file(*inputs, file_root, bytes_per_sample_hint=0, file_list='', initial_fill='', lazy_init='',
-         num_shards=1, pad_last_batch=False, prefetch_queue_depth=1, preserve=False, random_shuffle=False,
-         read_ahead=False, seed=-1, shard_id=0, shuffle_after_epoch=False, skip_cached_images=False,
-         stick_to_shard=False, tensor_init_bytes=1048576, device=None):
 
+def file(file_root, file_filters=None):
     Pipeline._current_pipeline._reader = "labelReader"
-    #Output
+    # Output
     labels = []
+    if file_filters is None:
+        file_filters = ["*.jpg", "*.jpeg", "*.png", "*.bmp", "*.tif", "*.tiff", "*.pnm", "*.ppm", "*.pgm", "*.pbm", "*.jp2", "*.webp"]
```

Comment:
Where is file_filters passed?

Reply:
Rajy mentioned that we need to pass file_filters argument from python to the file reader to read only those files. While its not yet been implemented, we are adding the file_filters argument from DALI currently. In future this will be used.

Reply:
Removed the file_filters list from the function